### ISSUE TITLE
ElasticSearch Filter Plugin

### ISSUE THREAD

**Issue by vinaymaddi1269** [opensearch-project/OpenSearch#1976]
Search Elasticsearch for a previous log event and copy some fields from it into the current event. Below are two complete examples of how this filter might be used.

Currently this filter is supported in Elastic eco system.

Use Cases and more details are present at https://www.elastic.co/guide/en/logstash/current/plugins-filters-elasticsearch.html

**Comment by vinaymaddi1269** [comment]
I am looking for equivalent OpenSearch plugin just like the above ElasticSearch plugin that works with eS

**Comment by RajivTaori** [comment]
This should probably be filed for Data Prepper, since this is referring to a logstash filter plugin

**Comment by dlvenable** [comment]
This is different from Data Prepper. The Logstash Elasticsearch filter plugin will make a query to Elasticsearch and update the current event in Logstash with data from Elasticsearch.

I believe the ask here is to create and support an OpenSearch filter plugin for Logstash.

**Comment by prashanttct07** [comment]
yes this is write David, ask is to find/write an alternative of elasticsearch filter for logstash in opensearch.

**Comment by vinaymaddi1269** [comment]
> This is different from Data Prepper. The Logstash Elasticsearch filter plugin will make a query to Elasticsearch and update the current event in Logstash with data from Elasticsearch.
> 
> I believe the ask here is to create and support an OpenSearch filter plugin for Logstash.

Thats correct

**Comment by dlvenable** [comment]
Thank you for the clarification. I created https://github.com/opensearch-project/data-prepper/issues/953 to see if something like this would be useful within Data Prepper.

**Comment by anasalkouz** [comment]
@dlvenable Shall we close this one on behalf of https://github.com/opensearch-project/data-prepper/issues/953?

**Comment by dlvenable** [comment]
@anasalkouz , No. This issue should stay open because the requestor is asking for a new Logstash plugin. Data Prepper is a distinct tool from Logstash.

**Comment by anasalkouz** [comment]
@elfisher can you please look into this and decide the next step?

**Comment by elfisher** [comment]
Is this the same as as the input plugin for logstash? https://github.com/opensearch-project/opensearch-plugins/issues/121

**Comment by prashanttct07** [comment]
@elfisher yes, it is similar only difference is that belongs to input of logstash where as this requirement is for filter section of logstash.

**Comment by elfisher** [comment]
@prashanttct07 thanks for clarifying. I'll add this to the backlog, and we will work to prioritize it. If this is something you (or anyone else) would be interested in contributing, let us know. We are happy to help review.

**Comment by gsinghk81** [comment]
Hi , Can we have the timelines for the availability of filter plugin

**Comment by dblock** [comment]
@gsinghk81 unless someone says otherwise nobody is currently working on this, so it doesn't have a timeline
